src/Common/src/Interop/Unix/System.Native/Interop.MountPoints.cs

```diff
@@ -0,0 +1,303 @@
+            ntfs = 0x5346544e,
+            openprom = 0x9fa1,
+            proc = 0x9fa0,
+            qnx4 = 0x002f,
+            reiserfs = 0x52654973,
+            romfs = 0x7275,
+            smb = 0x517B,
+            sysv2 = 0x012FF7B6,
+            sysv4 = 0x012FF7B5,
+            tmpfs = 0x01021994,
+            udf = 0x15013346,
+            ufs = 0x00011954,
+            usbdevice = 0x9fa2,
+            vxfs = 0xa501FCF5,
+            xenix = 0x012FF7B4,
+            xfs = 0x58465342,
+            xiafs = 0x012FD16D,
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct MountPointInformation
+        {
+            internal long                   AvailableFreeSpace;
+            internal long                   TotalFreeSpace;
+            internal long                   TotalSize;
+        }
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private unsafe delegate void MountPointFound(byte* name, int nameLength);
+
+        [DllImport(Libraries.SystemNative)]
+        private static extern int GetAllMountPoints(MountPointFound mpf);
+
+        [DllImport(Libraries.SystemNative)]
+        internal static extern int GetSpaceInfoForMountPoint([MarshalAs(UnmanagedType.LPStr)]string name, out MountPointInformation mpi);
+
+        [DllImport(Libraries.SystemNative)]
+        private unsafe static extern int GetFormatInfoForMountPoint(
+            [MarshalAs(UnmanagedType.LPStr)]string name,
+            byte* formatNameBuffer,
+            int bufferLength,
+            long* formatType);
+
+        internal static int GetFormatInfoForMountPoint(string name, out string format)
+        {
+            DriveType temp;
+            return GetFormatInfoForMountPoint(name, out format, out temp);
+        }
+
+        internal static int GetFormatInfoForMountPoint(string name, out DriveType type)
+        {
+            string temp;
+            return GetFormatInfoForMountPoint(name, out temp, out type);
+        }
+
+        private static int GetFormatInfoForMountPoint(string name, out string format, out DriveType type)
+        {
+            unsafe
+            {
+                byte* formatBuffer = stackalloc byte[MountPointFormatBufferSizeInBytes];    // format names should be small
+                long numericFormat;
+                int result = GetFormatInfoForMountPoint(name, formatBuffer, MountPointFormatBufferSizeInBytes, &numericFormat);
+                if (result == 0)
+                {
+                    // Check if we have an numeric answer or string
+                    if (numericFormat != -1)
+                    {
+                        format = Enum.GetName(typeof(LinuxFileSystemTypes), numericFormat);
+                        type = GetDriveType(format);
+                    }
+                    else
+                    {
+                        format = Marshal.PtrToStringAnsi((IntPtr)formatBuffer);
+                        type = GetDriveType(format);
```

Comment:
You can move this ```type = GetDriveType(format)``` to after the if/else, as it's the same line at the end of both blocks.